### DISCUSSION TITLE
Refresh Data Source

### DISCUSSION THREAD

**Discussion by ecxs** [Budibase/budibase#9328] [Help]
I'm creating a record via a side panel, and after the action 'Save Row', I've got a 'Execute Query' actions that runs the following query:

`UPDATE Assets SET Name = concat ( 'CBV-', YEAR(CURDATE()), id_sql ) WHERE id_sql = {{id}}`

This works fine, however, the Name is not immediately visible in the table. It is however, there, if I click on the row to check the details ...

![Screenshot 2023-01-12 at 14 25 02](https://user-images.githubusercontent.com/4005405/212078596-322f71f4-3b56-458d-b607-73cec5ee021d.png)

Please note that the table's data provider is sorted descending by the id, so the newly created record is the first record that is displayed.

**Comment by melohagan** [comment]
Hey @ecxs 

Have you tried using the **Refresh Data provider** action after your **Execute Query** action?

**Reply by ecxs** [reply]
Yes, I did.

**Reply by melohagan** [reply]
It must be related to this issue then: https://github.com/Budibase/budibase/issues/6872

I.e. the Refresh action must occur before the query has finished executing.
